core/crypto/crypto.go

```diff
@@ -64,16 +64,13 @@ type Client interface {
 	GetEnrollmentCertificateHandler() (CertificateHandler, error)
 
 	// GetTCertHandlerNext returns a CertificateHandler whose certificate is the next available TCert
-	GetTCertificateHandlerNext() (CertificateHandler, error)
+	GetTCertificateHandlerNext(attributes map[string]string) (CertificateHandler, error)
 
 	// GetTCertHandlerFromDER returns a CertificateHandler whose certificate is the one passed
-	GetTCertificateHandlerFromDER(der []byte) (CertificateHandler, error)
-
-	// ReadAttribute reads the attribute with name 'attributeName' from the der encoded x509.Certificate 'tcertder'.
-	ReadAttribute(attributeName string, tcertder []byte) ([]byte, error)
+	GetTCertificateHandlerFromDER(tCertDER []byte) (CertificateHandler, error)
 
 	// GetNextTCert gets next available (not yet used) transaction certificate.
-	GetNextTCert() (tCert, error)
+	GetNextTCert(attributes map[string]string) (tCert, error)
```

Comment:
Shall we avoid to expose internal interfaces like tCert? The corresponding public interface is CertificateHandler.